core/queryalgebra/evaluation/src/test/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/FilterOptimizerTest.java

```diff
@@ -75,8 +77,9 @@ public void deMerge() {
 		Filter spo = new Filter(new StatementPattern(s, p, o), oSmallerThanTwo);
 		Compare o2SmallerThanFour = new Compare(o2, four, CompareOp.LT);
 		Filter spo2 = new Filter(new StatementPattern(s, p, o2), o2SmallerThanFour);
-		TupleExpr expected = new Projection(new Join(spo, spo2), new ProjectionElemList(new ProjectionElem("s"),
-				new ProjectionElem("p"), new ProjectionElem("o"), new ProjectionElem("o2")));
+		TupleExpr expected = new QueryRoot(
```

Comment:
This change does not seem related to the thread-safety changes - accidentally left in?

Reply:
Yes the first commit in the series should not be there. I will try to fix and force push this.